sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -1605,4 +1625,292 @@ public String generateSas(BlobServiceSasSignatureValues blobServiceSasSignatureV
+
+            if (checkParametersNotNull(total)) {
+                AvroSchema.checkType("total", total, Long.class);
+                progressReceiver.reportProgress((Long) total);
+            } else {
+                return Mono.error(new IllegalArgumentException("Failed to parse end record from query "
+                    + "response stream."));
+            }
+        }
+        return Mono.empty();
+    }
+
+    /**
+     * Parses a quick query progress record.
+     * @param progressRecord The quick query progress record.
+     * @param progressReceiver The progress receiver.
+     * @return Mono.empty or Mono.error
+     */
+    private Mono<ByteBuffer> parseProgress(Map<?, ?> progressRecord, ProgressReceiver progressReceiver) {
+        if (progressReceiver != null) {
+            Object bytesScanned = progressRecord.get("bytesScanned");
+
+            if (checkParametersNotNull(bytesScanned)) {
+                AvroSchema.checkType("bytesScanned", bytesScanned, Long.class);
+                progressReceiver.reportProgress((Long) bytesScanned);
+            } else {
+                return Mono.error(new IllegalArgumentException("Failed to parse progress record from "
+                    + "query response stream."));
+            }
+        }
+        return Mono.empty();
+    }
+
+    /**
+     * Parses a quick query error record.
+     * @param errorRecord The quick query error record.
+     * @param errorReceiver The error receiver.
+     * @return Mono.empty or Mono.error
+     */
+    private Mono<ByteBuffer> parseError(Map<?, ?> errorRecord, ErrorReceiver<BlobQueryError> errorReceiver) {
+        Object fatal = errorRecord.get("fatal");
+        Object name = errorRecord.get("name");
+        Object description = errorRecord.get("description");
+        Object position = errorRecord.get("position");
+
+        if (checkParametersNotNull(fatal, name, description, position)) {
+            AvroSchema.checkType("fatal", fatal, Boolean.class);
+            AvroSchema.checkType("name", name, String.class);
+            AvroSchema.checkType("description", description, String.class);
+            AvroSchema.checkType("position", position, Long.class);
+
+            BlobQueryError error = new BlobQueryError((Boolean) fatal, (String) name,
+                (String) description, (Long) position);
+
+            if (errorReceiver != null) {
+                errorReceiver.reportError(error);
+            } else {
+                return Mono.error(new IOException("An error was reported during query response processing, "
+                        + System.lineSeparator() + error.toString()));
+            }
+        } else {
+            return Mono.error(new IllegalArgumentException("Failed to parse error record from "
+                + "query response stream."));
+        }
+        return Mono.empty();
+    }
+
+    /**
+     * Validates that all parameters are non-null. Throws IOException if any of them are.
```

Comment:
This doesn't seem to throw anything.

Reply:
That's poor porting on my part. It used to throw. Will fix the doc